### PR TITLE
add flowercare plugin

### PR DESCRIPTION
**Pull request by mzanetti**

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -139,6 +139,23 @@ Description: nymea.io plugin for elro
...
+ The nymea daemon is a plugin based IoT (Internet of Things) server. The
+ server works like a translator for devices, things and services and
+ allows them to interact.
+ With the powerful rule engine you are able to connect any device available
+ in the system and create individual scenes and behaviors for your environment.
+ .
+ This package will install the nymea.io plugin for Xiaomi Flower care devices,
+ also know as Plant care or MiCare
+
+
```
Comment (t-mon) [inline debian/control]: you should add this to the meta packages `nymea-plugins` and `nymea-plugins-all` too

File: flowercare/devicepluginflowercare.cpp [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+void DevicePluginFlowercare::postSetupDevice(Device *device)
+{
+    FlowerCare *flowerCare = m_list.value(device);
+    flowerCare->refreshData();
+}
+
+void DevicePluginFlowercare::deviceRemoved(Device *device)
+{
+    Q_UNUSED(device)
```
Comment (t-mon) [inline flowercare/devicepluginflowercare.cpp]: not unused any more

File: flowercare/devicepluginflowercare.cpp [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+    return false;
+}
+
+void DevicePluginFlowercare::onPluginTimer()
+{
+
+    foreach (FlowerCare *flowerCare, m_list) {
+        if (--m_refreshMinutes[flowerCare] <= 0) {
+            qCDebug(dcFlowerCare()) << "refreshing" << flowerCare->btDevice()->address();
```
Comment (t-mon) [inline flowercare/devicepluginflowercare.cpp]: Capital, just for the eye :)

File: flowercare/devicepluginflowercare.cpp [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+            flowerCare->refreshData();
+        } else {
+            qCDebug(dcFlowerCare()) << "not refreshing" << flowerCare->btDevice()->address() << " Next refresh in" << m_refreshMinutes[flowerCare] << "minutes";
```
Comment (t-mon) [inline flowercare/devicepluginflowercare.cpp]: Capital, just for the eye :)

File: flowercare/devicepluginflowercare.json [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                            "type": "QString",
+                            "inputType": "MacAddress"
+                        }
+                    ],
+                    "stateTypes": [
+                        {
+                            "id": "6254103a-ff8b-4f62-afda-6b5be1c933ec",
+                            "name": "connected",
+                            "displayName": "reachable",
+                            "displayNameEvent": "ronnected changed",
```
Comment (t-mon) [inline flowercare/devicepluginflowercare.json]: We should write these starting with a capital character and it should be `Reachable changed`

Comment (mzanetti) [reply flowercare/devicepluginflowercare.json]: Which looks quite crappy in nymea:app tbh

Comment (mzanetti) [reply flowercare/devicepluginflowercare.json]: which again shows that this is fundamentally wrong to have translations in the plugins themselves...

**Comment by mzanetti** [comment]
Jenkins, land this please

**Pull request merged**